schema.sql

```diff
@@ -19,12 +19,36 @@ CREATE TABLE species (
   id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
   name VARCHAR(100) NOT NULL );
 
+CREATE TABLE vets (
+  id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  name VARCHAR(100) NOT NULL,
+  age INT,
+  date_of_graduation DATE );
+
+CREATE TABLE specializations (
+  sid INT REFERENCES species(id),
+  vid INT REFERENCES vets(id),
+  PRIMARY KEY (sid, vid),
+  CONSTRAINT fk_specializations
+    FOREIGN KEY(sid) REFERENCES species(id) ON DELETE CASCADE,
+    FOREIGN KEY(vid) REFERENCES vets(id) ON DELETE CASCADE);
+
+CREATE TABLE visits (
+  aid INT,
+  vid INT, 
```

Comment:
[OPTIONAL ]
- `aid` and `vid` are not good column names. I sugges you use something like `animal_id` and `vet_id`. They make them easily read and identified when the database grows.